Redefinition of typedefs in ur_api.h

# Issue

The [ur_api.h](https://github.com/oneapi-src/unified-runtime/blob/main/include/ur_api.h) header redefines typdefs. Redefinition of typdefs is a C11 feature. Presumably we want to support earlier C versions.

# Example

To reproduce, compile the [ur_api.h](https://github.com/oneapi-src/unified-runtime/blob/main/include/ur_api.h) header on linux with `-Werror, -Wtypedef-redefinition`, this should give something like:
```
ur_api.h:274:39: error: redefinition of typedef 'ur_base_properties_t' is a C11 feature [-Werror,-Wtypedef-redefinition]                       
typedef struct _ur_base_properties_t ur_base_properties_t;                                                                                                                                   
                                      ^                                                                                                                                                        
ur_api.h:241:3: note: previous definition is here                                                                                               
} ur_base_properties_t; /source/ur/external/unified-runtime/include/ur_api.h:278:33: error: redefinition of typedef 'ur_base_desc_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
typedef struct _ur_base_desc_t ur_base_desc_t;
                                ^
ur_api.h:250:3: note: previous definition is here
} ur_base_desc_t;
  ^
ur_api.h:282:35: error: redefinition of typedef 'ur_rect_offset_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
typedef struct _ur_rect_offset_t ur_rect_offset_t;
                                  ^
ur_api.h:260:3: note: previous definition is here
} ur_rect_offset_t;
  ^
ur_api.h:286:35: error: redefinition of typedef 'ur_rect_region_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
typedef struct _ur_rect_region_t ur_rect_region_t;
                                  ^
ur_api.h:270:3: note: previous definition is here
} ur_rect_region_t;
  ^
ur_api.h:1702:3: error: redefinition of typedef 'ur_image_format_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
} ur_image_format_t;
  ^
ur_api.h:290:36: note: previous definition is here
typedef struct _ur_image_format_t ur_image_format_t;
                                   ^
ur_api.h:1720:3: error: redefinition of typedef 'ur_image_desc_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
} ur_image_desc_t;
  ^
ur_api.h:294:34: note: previous definition is here
typedef struct _ur_image_desc_t ur_image_desc_t;
                                 ^
ur_api.h:1848:3: error: redefinition of typedef 'ur_buffer_region_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
} ur_buffer_region_t;
  ^
ur_api.h:298:37: note: previous definition is here
typedef struct _ur_buffer_region_t ur_buffer_region_t;
                                    ^
ur_api.h:2211:3: error: redefinition of typedef 'ur_sampler_property_value_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
} ur_sampler_property_value_t;
  ^
ur_api.h:302:46: note: previous definition is here
typedef struct _ur_sampler_property_value_t ur_sampler_property_value_t;
                                             ^
ur_api.h:2816:3: error: redefinition of typedef 'ur_device_partition_property_value_t' is a C11 feature [-Werror,-Wtypedef-redefinition]
} ur_device_partition_property_value_t;
  ^
ur_api.h:306:55: note: previous definition is here
typedef struct _ur_device_partition_property_value_t ur_device_partition_property_value_t;
```

# Task

Remove the redefinition typdefs for the above types.

--- a/include/ur_api.h
+++ b/include/ur_api.h
@@ -269,43 +269,6 @@ typedef struct _ur_rect_region_t
 
 } ur_rect_region_t;
 
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_base_properties_t
-typedef struct _ur_base_properties_t ur_base_properties_t;
-
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_base_desc_t
-typedef struct _ur_base_desc_t ur_base_desc_t;
-
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_rect_offset_t
-typedef struct _ur_rect_offset_t ur_rect_offset_t;
-
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_rect_region_t
-typedef struct _ur_rect_region_t ur_rect_region_t;
-
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_image_format_t
-typedef struct _ur_image_format_t ur_image_format_t;
-
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_image_desc_t
-typedef struct _ur_image_desc_t ur_image_desc_t;
-
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_buffer_region_t
-typedef struct _ur_buffer_region_t ur_buffer_region_t;
-
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_sampler_property_value_t
-typedef struct _ur_sampler_property_value_t ur_sampler_property_value_t;
-
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ur_device_partition_property_value_t
-typedef struct _ur_device_partition_property_value_t ur_device_partition_property_value_t;
-
-
 #if !defined(__GNUC__)
 #pragma endregion
 #endif

--- a/scripts/templates/api.h.mako
+++ b/scripts/templates/api.h.mako
@@ -132,16 +132,6 @@ typedef struct _${th.subt(n, tags, obj['name'])} *${th.subt(n, tags, obj['name']
 
 %endif  # not re.match(r"class", obj['type'])
 %endfor # obj in spec['objects']
-## FORWARD-DECLARE STRUCTS ####################################################
-%if re.match(r"common", spec['name']):
-%for obj in th.extract_objs(specs, 'struct'):
-///////////////////////////////////////////////////////////////////////////////
-/// @brief Forward-declare ${th.make_type_name(n, tags, obj)}
-typedef struct _${th.make_type_name(n, tags, obj)} ${th.make_type_name(n, tags, obj)};
-
-%endfor
-
-%endif
 %if len(spec['objects']):
 #if !defined(__GNUC__)
 #pragma endregion

